### PR TITLE
apply partially the upcoming unnecessary_lambdas

### PR DESCRIPTION
**Pull request by a14n**
I haven't applied changes for codes like:

- [`setState`](https://github.com/flutter/flutter/blob/f9ad230f151408af5eb85229997d426662315609/packages/flutter/lib/src/material/date_picker.dart#L395-L397)
- [`setUpAll`](https://github.com/flutter/flutter/blob/f9ad230f151408af5eb85229997d426662315609/packages/flutter_tools/test/channel_test.dart#L15-L17)
- [`expect`](https://github.com/flutter/flutter/blob/f9ad230f151408af5eb85229997d426662315609/packages/flutter/test/foundation/change_notifier_test.dart#L190-L191)

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/daemon.dart [hunk 1]
```diff
@@ -264,7 +264,7 @@ class DaemonDomain extends Domain {
   }
 
   Future<Null> shutdown(Map<String, dynamic> args) {
-    Timer.run(() => daemon.shutdown());
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/daemon.dart]: is daemon final?

**Comment by Hixie** [comment]
LGTM

It's a pity this will likely regress over time.

File: packages/flutter_tools/lib/src/commands/daemon.dart (hunk 1 above)
Comment (a14n) [inline packages/flutter_tools/lib/src/commands/daemon.dart]: yes https://github.com/a14n/flutter/blob/ac9455f643a3a7ba3209374e8eb2c301a28cefea/packages/flutter_tools/lib/src/commands/daemon.dart#L156

**Comment by a14n** [comment]
This would be great to be able to configure exclusion of some methods for this linter rule. It'd allow to add the rule (and its exclusions) to `.analysis_options_repo`.

/cc @pq

**Comment by pq** [comment]
@a14n 👍  : could you open an issue to track this?  Thanks!

**Pull request merged**